gui/src/view.cc

```diff
@@ -785,6 +774,78 @@ void View::removeFromToolbar(GtkToolItem* ti)
   }
 }
 
+void View::setPanning()
+{
+  panning = true;
+}
+
+void View::unsetPanning()
+{
+  panning = false;
+}
+
+void View::registerSelectionListener(SelectionListener::Ptr listener)
+{
+  selectionListeners.push_back(listener);
+}
+
+void View::registerPostRenderer(PostRenderer::Ptr renderer)
+{
+  postRenderers.push_back(renderer);
+}
+
+void View::setStatusMessage(const std::string& message)
+{
+  gtk_statusbar_pop(statusBar, statusBarContextId);
+  gtk_statusbar_push(statusBar, statusBarContextId, message.c_str());
+}
+
+PresentationInterface::Ptr View::getCurrentPresentation()
+{
+  return presentation;
+}
+
+static void plugin_button_toggled(GtkToggleButton* button, gpointer data){
+  ToolStateListener* listener = static_cast<ToolStateListener*>(data);
+  if(gtk_toggle_button_get_active(button))
+  {
+    listener->onEnable();
+  }
+  else
+  {
+    listener->onDisable();
+  }
+}
+
+void View::addToolButton(const std::string& name, ToolStateListener::Ptr callback)
```

Comment:
You really want to add a GtkWidget or button or whatever here. Names are ugly and should (in time) be replaced by icons, which should be under control of the plugin

Reply:
Fair enough, this does pose a bit of a problem though, but perhaps this is only because of my limited experience with GTK+2.

- As far as I could tell the only button types available in GTK+2 that support button groups are the `GtkRadioButton` and `GtkRadioToolButton`. 
- As far as I can tell from the documentation there is no way to create a new `GtkRadioButton` with some other `GtkWidget` in/next to it.
- I did not manage to get any custom icons to show the normal way in a `GtkRadioButton`. 
- I did manage to get an icon/label to show in a `GtkRadioToolButton`, however this functionality seems to be limited to stock items in GTK+2 and they are size wise rather large (see image) in addition it's not really that clear when it is toggled, though that might have been because I did not hook it up to anything.
![afbeelding](https://user-images.githubusercontent.com/8530896/83916318-45a9f580-a775-11ea-8ee7-b6b5daed12d1.png)

Assuming that I am not missing something here then the only alternative implementations I can think of are:
- Adding both a `GtkRadioButton` and some `GtkWidget` supplied by the plugin on each tool button add which feels a bit hacky. 
- The `GtkRadioToolButton` seems promising, but also really limited (on first sight anyway). In addition stock items were deprecated in GTK3 with seemingly no replacement. 
- It would also technically be possible to implement our own button group logic if some other button type is more promising.

If you have anything in mind that I missed that allows both plugins to control what is shown and the view to managed the mutually exclusive aspect then please let me know @kees-jan. As it stands, I'm not entirely sure what the best course of action would be.

Reply:
I should also note that radio buttons with text don't really look all that great either due to it being hard to tell which radio button belongs to which label. Though I wasn't really able to find any better alternatives (so far).

Reply:
Didn't really dive into the technical details, but I'd say we're aiming for a look very similar to gimp, since that's what users will be expecting
![image](https://user-images.githubusercontent.com/221402/83968667-19a08880-a8cb-11ea-8faf-cd205f771da6.png)

They manage to make it clear which tool is selected, and use non-stock icons. Fortunately, the code is available :wink:

Reply:
That's a good idea, I'll look into the GIMP source, hopefully they did not implement something custom for this.

Reply:
Seems like GIMP both has some nice wrappers for this and is on GTK3 where there's a constructor for buttons with an image.

I did however finally get button images to work (turns out labels override images even if the image is set after the label).
![afbeelding](https://user-images.githubusercontent.com/8530896/84042750-c2bbb180-a9a5-11ea-8345-d98ec66f06f9.png)
So my plan for now is allow plugins to pass a `GtkToggleButton` or any of its sub classes and then implement button grouping manually. A plugin could then either return a toggle button with a text label or an image. Let me know if you think this is an acceptable solution.

Reply:
Pushed a commit for this. I'm currently not using an image for the measure tool because I don't really have a good image, but assuming it's not difficult to load/embed a png file (or raw image data I guess) in the plugin binary this wouldn't be too hard to add. I found that 16x16 pixels works best display wise.

There's also one last small 'issue' left. Currently disabling a plugin doesn't clear the status bar. I'm not sure whether to consider this a feature, responsibility of the plugin to clean up or responsibility of the view to clean up.

Reply:
Excellent. Wouldn't worry (at least for now) about text remaining in the statusbar.